client/components/common/ConfirmationModal.vue

```diff
@@ -36,13 +36,19 @@ export default {
     },
     close() {
       this.show = false;
-      this.context = createContext();
     },
     confirm() {
       this.context.action();
       this.close();
     }
   },
+  watch: {
+    show(val) {
+      const action = val ? this.context.onOpen : this.context.onClose;
+      if (!val) this.context = createContext();
+      return action && action();
```

Comment:
why not keeping the initial version and adding onOpen onClose in open/close methods ? Seems more readable (no if for context reset, no action resolution) and does not create extra watcher 🤔

Reply:
I was going to do this initially then I realised clicking outside does not invoke close method.

Reply:
Nvm. Handled this with vuetify dialog click:outside event. 🤡